src/test/java/com/example/bookstore/controller/CategoryControllerTest.java

```diff
@@ -0,0 +1,114 @@
+import org.springframework.web.context.WebApplicationContext;
+
+@SpringBootTest(webEnvironment = WebEnvironment.RANDOM_PORT)
+public class CategoryControllerTest {
+
+    protected static MockMvc mockMvc;
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    @BeforeAll
+    static void beforeAll(@Autowired DataSource dataSource,
+            @Autowired WebApplicationContext applicationContext) throws SQLException {
+        mockMvc = MockMvcBuilders
+            .webAppContextSetup(applicationContext)
+            .apply(SecurityMockMvcConfigurers.springSecurity())
+            .build();
+        teardown(dataSource);
+        try (Connection connection = dataSource.getConnection()) {
+            connection.setAutoCommit(true);
+            ScriptUtils.executeSqlScript(
+                    connection,
+                  new ClassPathResource("database/books/add-three-default-books.sql"));
+            ScriptUtils.executeSqlScript(
+                    connection,
+                  new ClassPathResource("database/books/add-three-default-categories.sql"));
+            ScriptUtils.executeSqlScript(
+                    connection,
+                  new ClassPathResource("database/books/add-categories-for-books.sql"));
+        }
+    }
+
+    @AfterAll
+    static void afterAll(@Autowired DataSource dataSource) {
+        teardown(dataSource);
+    }
+
+    @SneakyThrows
+    static void teardown(DataSource dataSource) {
+        try (Connection connection = dataSource.getConnection()) {
+            connection.setAutoCommit(true);
+            ScriptUtils.executeSqlScript(
+                    connection,
+                  new ClassPathResource("database/books/remove-all-books.sql"));
+            ScriptUtils.executeSqlScript(
+                    connection,
+                  new ClassPathResource("database/books/remove-all-categories.sql"));
+            ScriptUtils.executeSqlScript(
+                    connection,
+                  new ClassPathResource("database/books/remove-all-books-categories.sql"));
+        }
+    }
+
+    @Test
+    @WithMockUser(username = "admin", roles = {"ADMIN"})
+    public void createCategory_ShouldReturnCreatedCategory() throws Exception {
+        // Arrange
+        CategoryRequestDto requestDto = new CategoryRequestDto("Science", "Books about science");
+        String requestBody = objectMapper.writeValueAsString(requestDto);
+
+        // Act & Assert
+        mockMvc.perform(post("/api/categories")
+                .contentType(MediaType.APPLICATION_JSON)
+                .content(requestBody))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$.name").value("Science"))
+                .andExpect(jsonPath("$.description").value("Books about science"));
+
+        // Teardown
+        mockMvc.perform(delete("/api/categories/" + 4) // Assuming the new category ID is 4
+                .contentType(MediaType.APPLICATION_JSON))
```

Comment:
remove comments. Don’t use separate endpoints in one test case. Make your test cases isolated